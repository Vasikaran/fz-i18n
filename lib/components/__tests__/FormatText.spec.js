'use strict';

var _FormatText = require('../FormatText');

var _FormatText2 = _interopRequireDefault(_FormatText);

var _I18NProvider = require('../I18NProvider');

var _I18NProvider2 = _interopRequireDefault(_I18NProvider);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('FormatText component', function () {
  it("Should display i18n value as html", function () {
    var ele = _reactTestRenderer2.default.create(_react2.default.createElement(
      _I18NProvider2.default,
      { i18n: { key1: "vimal1<b>vimal</b>" } },
      _react2.default.createElement(_FormatText2.default, { i18NKey: 'key1', isHtml: true })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should display i18n value", function () {
    var ele = _reactTestRenderer2.default.create(_react2.default.createElement(
      _I18NProvider2.default,
      { i18n: { key1: "vimal1<b>vimal</b>" } },
      _react2.default.createElement(_FormatText2.default, { i18NKey: 'key1' })
    ));
    var tree = ele.toJSON();
    expect(tree).toMatchSnapshot();
  });
});