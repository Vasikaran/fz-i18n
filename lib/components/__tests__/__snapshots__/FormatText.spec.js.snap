exports[`FormatText component Should display i18n value 1`] = `
<span>
  vimal1<b>vimal</b>
</span>
`;

exports[`FormatText component Should display i18n value as html 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "vimal1<b>vimal</b>",
    }
  } />
`;
